src/app/containers/OptimizelyArticleCompleteTracking/index.test.jsx

```diff
@@ -0,0 +1,211 @@
+  onReady: jest.fn(() => Promise.resolve()),
+  track: jest.fn(),
+};
+
+const observers = new Map();
+
+const IntersectionObserver = jest.fn(cb => {
+  const item = {
+    callback: cb,
+    elements: new Set(),
+  };
+
+  const instance = {
+    observe: jest.fn(element => {
+      item.elements.add(element);
+    }),
+    disconnect: jest.fn(() => {
+      item.elements.clear();
+    }),
+  };
+
+  observers.set(instance, item);
+
+  return instance;
+});
+
+const getObserverInstance = element => {
+  try {
+    const [instance] = Array.from(observers).find(([, item]) =>
+      item.elements.has(element),
+    );
+
+    return instance;
+  } catch (e) {
+    throw new Error('Failed to find IntersectionObserver for element.');
+  }
+};
+
+const triggerIntersection = ({ changes, observer }) => {
+  const item = observers.get(observer);
+
+  item.callback(changes);
+};
+
+const ContextWrap = ({ pageType, isAmp, children, service }) => (
+  <RequestContextProvider
+    isAmp={isAmp}
+    pageType={pageType}
+    service={service}
+    pathname="/pathname"
+  >
+    <OptimizelyProvider optimizely={optimizely} isServerSide>
+      {children}
+    </OptimizelyProvider>
+  </RequestContextProvider>
+);
+
+ContextWrap.propTypes = {
+  children: node.isRequired,
+  pageType: string.isRequired,
+  isAmp: bool.isRequired,
+  service: string.isRequired,
+};
+
+const { error } = console;
+
+beforeEach(() => {
+  jest.clearAllMocks();
+  jest.useFakeTimers();
+  console.error = jest.fn();
+  global.IntersectionObserver = IntersectionObserver;
+});
+
+afterEach(() => {
+  jest.runOnlyPendingTimers();
+  jest.useRealTimers();
+  console.error = error;
+  observers.clear();
+});
+
+describe('Optimizely Page Complete tracking', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should return a function that can be assigned to an element to observe for intersections', () => {
+    const { container } = render(
+      <ContextWrap pageType={ARTICLE_PAGE} service="news" isAmp={false}>
+        <OptimizelyArticleCompleteTracking />
+      </ContextWrap>,
+    );
+
+    const element = container.getElementsByTagName('div')[0];
+    const { observe } = getObserverInstance(element);
+
+    expect(observe).toHaveBeenCalledWith(element);
+  });
+
+  it('should not send tracking event when element is not in view', async () => {
+    useOptimizelyVariation.mockReturnValue('variation_1');
+
+    const { container } = render(
+      <ContextWrap pageType={ARTICLE_PAGE} service="news" isAmp={false}>
+        <OptimizelyArticleCompleteTracking />
+      </ContextWrap>,
+    );
+
+    const element = container.getElementsByTagName('div')[0];
+    const observerInstance = getObserverInstance(element);
+
+    act(() => {
+      triggerIntersection({
+        changes: [{ isIntersecting: false }],
+        observer: observerInstance,
+      });
+    });
+
+    await waitFor(() => {
+      expect(optimizely.track).toHaveBeenCalledTimes(0);
+    });
```

Comment:
I think this test would always pass as it's a negative test. The expectation will instantly pass as it will have initially be called zero times. The only way I can think of to make a test like this work is to add an arbitrary timed wait but even that can be flawed in the logic takes a whole to execute. I'd suggest removing the test on that basis unless you can think of a different approach?